File: prog2/2. hét/szulo_gyerek.java
public class szulo_gyerek {
    
    public static void main(String[] args) {
        Szulo szulo = new Szulo(); //ősosztály Object osztály
        Gyerek gyerek = new Gyerek();
        Szulo ujgyerek= new Gyerek();
        //Gyerek ujszulo = new Szulo(); //Gyerek osztály több mindent tartalmaz a Szülőnél
        szulo.fuggveny();
        gyerek.fuggveny();
        ujgyerek.fuggveny();
        //ujgyerek.ujfuggveny();   //nem hívható meg a metódus
    }
}

    class Szulo{  //nem lehet publikus
        public void fuggveny(){
            System.out.println("szülő");
        };
    }
    
    class Gyerek extends Szulo{

        @Override
        public void fuggveny() {
            System.out.println("gyerek");
            //super.fuggveny(); //ősosztály függvény metódusa
        }
        public void ujfuggveny() {
            System.out.println("ujfuggveny");
        }
            
    }

File: prog2/2. hét/szulo_gyerek.cpp
#include <iostream>
using namespace std;

class Szulo{
    public:
        virtual void fuggveny(){
            cout<<"szulo"<<endl;
        };
};

class Gyerek: public Szulo{
    public:
        void fuggveny() {
            cout<<"gyerek"<<endl;
        }
        void ujfuggveny() {
            cout<<"ujfuggveny"<<endl;
        }
};

int main() {
    Szulo *szulo;
    szulo = new Szulo();

    Gyerek *gyerek;
    gyerek = new Gyerek();

    Szulo *ujgyerek;
    ujgyerek = new Gyerek();

    Gyerek *ujszulo;
    //ujszulo = new Szulo(); //Gyerek osztály több mindent tartalmaz a Szülőnél

    szulo->fuggveny();
    gyerek->fuggveny();
    ujgyerek->fuggveny();
    //ujgyerek->ujfuggveny(); //a Szülő classnak nincs ujfuggveny nevű függvénye
    
    return 0;
}
